prometheus/promhttp/http.go

```diff
@@ -418,6 +419,43 @@ type HandlerOpts struct {
 	ProcessStartTime time.Time
 }
 
+type OpenMetricsOptions struct {
```

Comment:
Hm.. want to play with different ideas for this options first 🤔 Even keeping `EnableOpenMetrics` and then `OpenMetricsOptions.AddV1CreatedSeries` might a better choice? 🤔

Reply:
> Hm.. want to play with different ideas for this options first 🤔 Even keeping `EnableOpenMetrics` and then `OpenMetricsOptions.AddV1CreatedSeries` might a better choice? 🤔

Interesting! I assume that this suggestion comes from not liking the idea of mentioning that "Created" is a "Series", this is something we'd like to change in the next version to something closer to metadata.

What do you think of `EnableCreatedTimestamps` instead? 

I think mentioning versions on fields is not the best, in the future I can see a `version` field that controls how each feature is exposed instead of `AddV1Created..`, `AddV2Created...` and then handling mutually exclusiveness

Reply:
Oh lol, I just realized I suggested what's already in the code 🤦

Reply:
I understand the confusion, here is why: https://github.com/prometheus/client_golang/pull/1408#discussion_r1774821420

We will likely not have "enable X Y Z" unless protocol will make a challenging decision like with CT being a series. So ideally we only have this "unsafe 1.0 Ct as series" flag for now, but do CT by default if they are safe to use (they don't increase 50-30% of Prometheus memory for no reason). (:

Reply:
We might have some more flags in future for units too though, so some plan for those would be nice. For CT as series that an OpenMetrics option, but for units (e.g. AutogenerateUnits)... it's kind of general option? maybe not even in promhttp but in registry or something?

Since we now little and we already make it complex I would vote for keeping `EnableOpenMetrics` and adding `EmitOpenMetricsTextCreatedSamples bool` would be perfect for now. Samples not series as this is how it's referenced in OpenMetrics  https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#counter-1

Simple, and effective, then we probably should fail or ignore this if EnableOpenMetrics is not true.

WDYT

Reply:
On a loosely relate note, I would agree on a general option for units. Happy to resume the work on that one whenever (if ever) needed.

Reply:
I'm still confused here haha

`EmitOpenMetricsTextCreatedSamples bool`, this means there's no distinction between OM v1 and v2, do I understand that correctly? If I'm following you correctly, if we introduce this flag now with `false` as the default value, we won't be able to change it once OM v2 is out, right?

Reply:
That's exactly why I call this option "CreatedSamples", so the not ideal _created lines. Once in V2 we have that in metadata or as annotation, even with this option false we will emit CT semantics by default IMO. 

This the same as with exemplars - enable open metrics, enables exemplars, no questions asked, because they were designed to NOT spam TSDB with odd series (:

Perhaps let's chat offline if this is still not clear/I am missing something? (: